### PR TITLE
Refix generation of thumbnails with inline thumbnail config

### LINKED ISSUE #16439: [Bug]: Thumbnail generation fails with image editable with cropping and cache buster
### Pimcore version

11.1.4

### Steps to reproduce

Hi, this project was updated from Pimcore 10 to Pimcore 11. There was no problem in Pimcore 10.

I've experienced an issue when using image editables **with** cropping and **cache buster active** in some cases. 

```
{{ pimcore_image('image', {
    thumbnail: 'teaser',
    cacheBuster: true,
}) }}
```

Example path frontend output:
```
/cache-buster-5/folder1/folder2/2147/image-thumb__2147__teaser_auto_8d1d280eddf50a7e0ad7b3d7a89b0635/filename@2x.aba9c031.jpg
```

In this case $thumbnailConfig is empty, which results in showing the fallback svg image (/bundles/pimcoreadmin/img/filetype-not-supported.svg)
https://github.com/pimcore/pimcore/blob/2b3bb4d10f4c32270589bb37557b932ca38d2a7e/models/Asset/Service.php#L509

After disabling the cachseBuster, the image shows up.

### Actual Behavior

-

### Expected Behavior

-

### PR DESCRIPTION
**Pull request by fashxp**
fix of #15198 had a problem, when different croppings of the same image were rendered in one page ...

see in demo 

first request: 
image

second request: 
image

third request: 
image

forth request (finally correct): 
image

problem was, that pathinfo doesn't contain all necessary information.

### CODE REVIEW

File: models/Asset/Image/Thumbnail/Processor.php [hunk 1]
```diff
@@ -215,7 +215,7 @@ public static function process(
         if ($deferred) {
             // only add the config to the TmpStore if necessary (e.g. if the config is auto-generated)
             if (!Config::exists($config->getName())) {
-                $pathInfo = trim($asset->getRealPath(), '/').'/'.$asset->getId() . '/';
+                $pathInfo = trim($asset->getRealPath(), '/').'/'.$asset->getId() . '/' . $config->getName();
```
Comment (brusch) [inline models/Asset/Image/Thumbnail/Processor.php]: Unfortunately this doesn't cover different image formats (when using auto, e.g. WebP, AVIF, JPG) as well as different resolutions (`@2x`). 
That's why we used to utilize the `$storagePath` here, which includes both variables.

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/Asset/Image/Thumbnail/Processor.php (hunk 1 above)
Comment (hadl) [reply models/Asset/Image/Thumbnail/Processor.php]: I can confirm that. We are heavily relying on auto thumbs for responsive images and with #15198 they are really broken.
Guess creating the deferred config id using only the basepath and asset id is not enough?

Simplest example is using 
```
{{ pimcore_image("myImage", {
    "thumbnail": {
        "width": 200,
        "height": 200
    }
}) }}
{{ pimcore_image("myImage2", {
    "thumbnail": {
        "width": 300,
        "height": 400
    }
}) }}
```
Then try to open an image from the sources only by path - eg. currently displayed is webp -> use jpg.

Using an asset saved in root (no folder) of the assets-tree is not working either.

Comment (fashxp) [reply models/Asset/Image/Thumbnail/Processor.php]: Filename is now added to deferred config id, and root-level assets are fixed too. 
Can you please double check.

Comment (hadl) [reply models/Asset/Image/Thumbnail/Processor.php]: root-level assets work -> lgtm

I see that there are more TmpStore entries for the thumb configs -> lgtm

I have the same asset twice on the page with different aspect-ratios and methods (cover/frame) -> lgtm
Selecting an other image from the src-set seems to work -> lgtm


But it looks like the thumbnails are only created/served correctly at the second request.

Found this in dev.log. Hope this can help.
```
1. Request:

[2024-01-17T10:26:38.784925+01:00] request.INFO: Matched route "_pimcore_service_thumbnail". {"route":"_pimcore_service_thumbnail","route_parameters":{"_route":"_pimcore_service_thumbnail","_controller":"Pimcore\\Bundle\\CoreBundle\\Controller\\PublicServicesController::thumbnailAction","prefix":"18/","type":"image","assetId":"18","thumbnailName":"_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca","filename":"4-542-928-0001-DYN-Katalog.b6085414.webp"},"request_uri":"http://xxxxx.test/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","method":"GET"} []

[2024-01-17T10:26:38.785701+01:00] app.DEBUG: Resolved pimcore context for path /18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp to default {"path":"/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","context":"default"} []

[2024-01-17T10:26:38.808707+01:00] pimcore.ERROR: Unable to generate image thumbnail, see logs for details. [] []

=====
2. Request

[2024-01-17T10:36:04.645148+01:00] request.INFO: Matched route "_pimcore_service_thumbnail". {"route":"_pimcore_service_thumbnail","route_parameters":{"_route":"_pimcore_service_thumbnail","_controller":"Pimcore\\Bundle\\CoreBundle\\Controller\\PublicServicesController::thumbnailAction","prefix":"18/","type":"image","assetId":"18","thumbnailName":"_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca","filename":"4-542-928-0001-DYN-Katalog.b6085414.webp"},"request_uri":"http://xxxxx.test/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","method":"GET"} []

[2024-01-17T10:36:04.645320+01:00] app.DEBUG: Resolved pimcore context for path /18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp to default {"path":"/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","context":"default"} []

[2024-01-17T10:36:04.650961+01:00] init.DEBUG: Resolved editmode to false {"editmode":"false","params":{"param":false,"adminRequest":false,"user":false}} []

[2024-01-17T10:36:04.654153+01:00] php.DEBUG: Warning: filemtime(): stat failed for /xxxxx/public/var/tmp/thumbnails/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp {"exception":{"Symfony\\Component\\ErrorHandler\\Exception\\SilencedErrorContext":{"severity":2,"file":"/xxxxx/vendor/league/flysystem-local/LocalFilesystemAdapter.php","line":427,"trace":[{"file":"/xxxxx/vendor/league/flysystem/src/Filesystem.php","line":161,"function":"lastModified","class":"League\\Flysystem\\Local\\LocalFilesystemAdapter","type":"->"}],"count":1}}} []

[2024-01-17T10:36:04.817648+01:00] pimcore.DEBUG: Thumbnail /18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp generated in 0.162113904953 seconds [] []
```

**Review by brusch — APPROVED** [review]

**Comment by brusch** [comment]
LGTM 👌

File: models/Asset/Image/Thumbnail/Processor.php (hunk 1 above)
Comment (hadl) [reply models/Asset/Image/Thumbnail/Processor.php]: Deleted all thumbnail files and open page:
SCR-20240117-kpgx

Comment (fashxp) [reply models/Asset/Image/Thumbnail/Processor.php]: Hmm strange. Works for me with tje first request too. Could you try to debug on your end?

Comment (hadl) [reply models/Asset/Image/Thumbnail/Processor.php]: I have logged in Thumbnail\Processor and Asset\Image Service on TmpStore::add and TmpStore::get.
On the first page load there is 1 "add" missing and then the "get" fails, of course.
Page reload then is adding the missing config and it works.

Strange thing is that this is exactly the image which is supposed to be visible (on desktop width).
In addition, there are 2 "image-blocks" on the page with the same asset and its either the first or the second failing after deleting all the thumbnails.

Log is with "$pathInfo | $configId"
```
[2024-01-17T13:09:53.198389+01:00] pimcore.INFO: TmpStore::add thumbnail_deferred: 18/_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.7979b474.png | thumb_18__449f187610c6fd50ebb6764c72f03a14 [] []

[2024-01-17T13:09:53.205981+01:00] pimcore.INFO: TmpStore::add thumbnail_deferred: 18/_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.7979b474.png | thumb_18__449f187610c6fd50ebb6764c72f03a14 [] []

[2024-01-17T13:09:53.641316+01:00] pimcore.INFO: TmpStore::get thumbnail_deferred FAILED: 18/_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp | thumb_18__3688dafc50a4f45cbf7eb712730267fd [] [] ❗️

====
page reload
===

[2024-01-17T13:23:02.622932+01:00] pimcore.INFO: TmpStore::add thumbnail_deferred: 18/_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.7979b474.png | thumb_18__449f187610c6fd50ebb6764c72f03a14 [] []

[2024-01-17T13:23:02.623487+01:00] pimcore.INFO: TmpStore::add thumbnail_deferred: 18/_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp | thumb_18__3688dafc50a4f45cbf7eb712730267fd [] [] ✅

[2024-01-17T13:23:02.629771+01:00] pimcore.INFO: TmpStore::add thumbnail_deferred: 18/_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.7979b474.png | thumb_18__449f187610c6fd50ebb6764c72f03a14 [] []
```

Guess i have to double-check our Twig-ImageExtension which handles the image `srcset`, too.

---
Bottom line: It works **way** better than before. A big **thank you** for your effort!

Comment (fashxp) [reply models/Asset/Image/Thumbnail/Processor.php]: hmm, I cannot reproduce that on my end. For me, the images work on the first load.
I would merge that now. If you have any additional findings, please feel free to create an issue, or even better already a PR to fix it ;-)

Comment (hadl) [reply models/Asset/Image/Thumbnail/Processor.php]: Just one question:
The expiration date for the TmpStore entries should be higher (or even off ) ?
With full_page_cache enabled it can / will happen that the entry is removed and then the auto thumb cannot be created anymore.

Comment (bitbirddev) [reply models/Asset/Image/Thumbnail/Processor.php]: > Just one question: The expiration date for the TmpStore entries should be higher (or even off ) ? With full_page_cache enabled it can / will happen that the entry is removed and then the auto thumb cannot be created anymore.

it gets worse when using frontend_prefixes (cdn) because the frontend_prefix is only added if the thumbnail is already created. would be nice if the fullpage cache only kicks in when all thumbnails are generated for that page. it would also be nice if we would  have a command that generates thumbnails for alle thumbs in the tmpstore...

**Comment by bitbirddev** [comment]
@fashxp there is another bug in the thumbnails in combination with `frontend_prefixes`

### To reproduce the Bug:

#### 1. Add Frontend Prefixes
```yaml
pimcore:
    assets:
        frontend_prefixes:
            source: "%env(CDN_DOMAIN)%/assets"
            thumbnail: "%env(CDN_DOMAIN)%/thumbnails"
```

#### 2. Clear/Truncate
- tmp_store
- assets_thumbnail_cache
- Temporary Files (via Button in Admin->Settings->Cache->Clear temporary files

#### 3. Create a template with an image-tag 
`{{ pimcore_image('test', {'thumbnail': {'width': 200, 'height': 200}}) | raw }}`



___we now have to make some request-roundtrips to get to the problem___

When you first open your template in the frontend, the image tag should look similar to this

```html
<picture>
	<source srcset="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.webp 1x, /3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage@2x.e347a3c5.webp 2x" width="200" height="200" type="image/webp">
	<source srcset="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.jpg 1x, /3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage@2x.e347a3c5.jpg 2x" width="200" height="200" type="image/jpeg">
	<img src="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.jpg" width="200" height="200" alt="" loading="lazy" srcset="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.jpg 1x, /3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage@2x.e347a3c5.jpg 2x">
</picture>
```

In my case my browser choose to load the .webp-Version of the srcset:

image

So after the first request ___ONE___ thumbnail is generated 

image


while the other 3 thumbnails are deferred and their configs saved to the tmp_store for later

image

everything is fine for now. now reload the page. as you can see, your picture-tag has changed. the thumbnail that was generated in the first request got his frontend_prefixes now and thus is loaded via S3 in my case.

```html
<picture>
	<source srcset="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.webp 1x, https://pimcore-boilerplate.s3.eu-central-1.amazonaws.com/thumbnails/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage@2x.e347a3c5.webp 2x" width="200" height="200" type="image/webp">
	<source srcset="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.jpg 1x, /3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage@2x.e347a3c5.jpg 2x" width="200" height="200" type="image/jpeg">
	<img src="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.jpg" width="200" height="200" alt="" loading="lazy" srcset="/3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage.e347a3c5.jpg 1x, /3/image-thumb__3___auto_ea796264b15fae954fcc6cb999b3b659/testimage@2x.e347a3c5.jpg 2x">
</picture>
```

your database tables didn't change, `tmp_store` and `assets_image_thumbnails_cache` look just as in the screenshot above

### The final part

- choose a URL of your srcset that wasn't requested from your browser (the ones without fronted_prefix)
- Right click and open the image in a new Tab (you should see the image)
- Now Reload the tab => ___THE IMAGE IS GONE___
SCR-20240118-bwhb


Not only the image is gone, also your `assets_image_thumbnails_cache` is cleared. 

***Since the `tmp_store`-Entry for this specific thumbnail was deleted after the thumbnail was created and since the corresponding entry in `assets_image_thumbnails_cache` is now also missing, the thumb can never be generated ever again.***


SCR-20240118-bxcs

And the worst part is that for this image the `assets_image_thumbnails_cache` won't ever get filled again because the files still exist on the disk

### Why is this Bad
- loading a deferred thumbnail url twice wipes out all your `assets_image_thumbnails_cache`-Entries for this image
- it also breaks this specific image-thumbnail
- when using the full page cache, chances are very high that an deferred thumbnail url is cached => high chances that this url is requested twice
- chances are high that you don't see your missing images because your browser requests another version of the thumb. your users might see broken images even everything seems ok on your side

you might first think this is an edge-case and doesn't really happen but it happens all the time. i've noticed this in production for a while now. don't forget that not only users visit our sites but also a lot of crawlers. 

### What else
- it would be nice to have a console command to generate thumbs for entries in the `tmp_store`
- the full page cache should only be activated if no deferred thumbnail urls are found in the response (if `frontend_prefixes` are enabled)

**Comment by bitbirddev** [comment]
also take a look at this related discussion #14615

**Comment by fashxp** [comment]
@bitbirddev please create another issue for that and maybe you also have already some ideas to fix it to provide a PR ... thank you very much...

File: models/Asset/Image/Thumbnail/Processor.php (hunk 1 above)
Comment (brusch) [reply models/Asset/Image/Thumbnail/Processor.php]: @hadl I wouldn't keep the TmpStore items forever, but yes we can think about extending the time or even make it configurable. Could you create a PR for that? 

@bitbirddev I see the challenge that then a page will never get into the cache, because there can be so many different variants of a thumbnail, different formats, different resolutions, media queries, ... don't think that this would make sense in practice. 😬

Comment (bitbirddev) [reply models/Asset/Image/Thumbnail/Processor.php]: @brusch please take a look at #16467 for a detailed description of the problem

**Comment by bitbirddev** [comment]
> @bitbirddev please create another issue for that and maybe you also have already some ideas to fix it to provide a PR ... thank you very much...

done. see #16486
